packages/react-next/src/components/OverflowSet/OverflowButton.base.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { KeytipManager } from 'office-ui-fabric-react/lib/utilities/keytips/KeytipManager';
+import { IKeytipProps } from '../../Keytip';
+import { IOverflowSetItemProps, IOverflowSetProps } from './OverflowSet.types';
+import { useConst } from '@uifabric/react-hooks';
+
+const useKeytipRegistrations = (
+  persistedKeytips: { [uniqueID: string]: IKeytipProps },
+  keytipManager: KeytipManager,
+) => {
+  React.useEffect(() => {
+    Object.keys(persistedKeytips).forEach((keytipId: string) => {
+      const keytip = persistedKeytips[keytipId];
+      const uniqueID = keytipManager.register(keytip!, true);
+      // Update map
+      persistedKeytips[uniqueID] = keytip;
+      delete persistedKeytips[keytipId];
+    });
+    return () => {
+      // Delete all persisted keytips saved
+      Object.keys(persistedKeytips).forEach((uniqueID: string) => {
+        keytipManager.unregister(persistedKeytips[uniqueID]!, uniqueID, true);
+        delete persistedKeytips[uniqueID];
+      });
+    };
+  }, [persistedKeytips, keytipManager]);
+};
+
+export const OverflowButtonBase = (props: IOverflowSetProps) => {
+  const keytipManager: KeytipManager = KeytipManager.getInstance();
+  const { className, overflowItems, keytipSequences } = props;
+  const persistedKeytips = useConst<{ [uniqueID: string]: IKeytipProps }>({});
+
+  // Gets the subMenu for an overflow item
+  const getSubMenuForItem = React.useCallback(
+    (item: IOverflowSetItemProps) => {
+      // Checks if itemSubMenuProvider has been defined, if not defaults to subMenuProps
+      if (props.itemSubMenuProvider) {
+        return props.itemSubMenuProvider(item);
+      }
+      if (item.subMenuProps) {
+        return item.subMenuProps.items;
+      }
+      return undefined;
+    },
+    [props],
```

Comment:
always put only what's absolutely required here to avoid unnecessary cache busting:
```
    [itemSubMenuProvider],
```